Ret ixmp

adrivinca opened the pull request:
- fix of the retixmp package that allow to load on ixmp any kind of dataframe and array
- added retixmp installation in intall.bat
- expanded the README, explaining the requirements for using retixmp

# Please confirm that this PR has done the following:

- [ ] Tests Added
- [ ] Documentation Added
- [ ] Description in RELEASE_NOTES.md Added

# Adding to RELEASE_NOTES.md

Please add a single line in the release notes similar to the following:

```
- [#69][https://github.com/iiasa/ixmp/pull/69] bring retixmp available to other users
```

gidden commented on ixmp/default_path_constants.py:
```diff
@@ -1,7 +1,8 @@
 # path to local files
-LOCAL_PATH = os.path.expanduser(os.path.join('~', '.local', 'ixmp'))
+LOCAL_PATH = os.path.expanduser(os.path.join('~', '.local', 'ixmp')
+                                ).replace('Documents\\', '')
```
can you make this please

```
LOCAL_PATH = os.path.expanduser(os.path.join('~', '.local', 'ixmp'))
LOCAL_PATH = LOCAL_PATH.replace('Documents' + os.path.sep, '') # fix for R users
```

adrivinca replied:
thanks, done like this!

gidden commented:
1 change requested. someone familiar on the R side should make sure all is well. @iiasa/messageix-r-devs

adrivinca commented on ixmp/default_path_constants.py:
```diff
@@ -2,6 +2,7 @@
 
 # path to local files
 LOCAL_PATH = os.path.expanduser(os.path.join('~', '.local', 'ixmp'))
+LOCAL_PATH = LOCAL_PATH.replace('Documents' + os.path.sep, '') # fix for R users
```
Ah come one Mr stickler! these two errors can't be solved together!! unless, we move the comment, but let's keep it an inline comment

gidden replied:
Probably easiest to move it up one line there (sorry!)

gidden commented:
https://github.com/spyder-ide/spyder-autopep8 may be helpful to catch these
before you hit the commit button, FYI!

On Tue, Jul 10, 2018 at 2:38 PM, Stickler Bot <notifications@github.com>
wrote:

> *@stickler-ci* commented on this pull request.
> ------------------------------
>
> In ixmp/default_path_constants.py
> <https://github.com/iiasa/ixmp/pull/69#discussion_r201325037>:
>
> > @@ -2,6 +2,8 @@
>
>  # path to local files
>  LOCAL_PATH = os.path.expanduser(os.path.join('~', '.local', 'ixmp'))
> +# fix for R users
> +LOCAL_PATH = LOCAL_PATH.replace('Documents' + os.path.sep, '')
>
> W291 trailing whitespace
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/iiasa/ixmp/pull/69#pullrequestreview-135802350>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABVAEVFqSSTDMMoei-OOvuEH8L_FTVraks5uFKA2gaJpZM4VI9WG>
> .
>

adrivinca commented on README.md:
```diff
@@ -123,7 +123,13 @@ the ix modeling platform for scientific publications or technical reports:
+
+0. Two alternative options are available:
+
+    - use the package developed in Python, requires the `reticulate` R package (it will allow to install/use `retixmp`)
```
Thanks for testing it, I have not problems when solving in the tutorial, maybe we can check it together.
Yes I can write somewhere the difference in syntax between the two packages.

the dependences on other packages are only written in the readme, not installed automatically, because installing automatically all the R packages might create more problems. 
It was the same for rixmp and rmessageix and their dependences, if something is missing the installation is skipped. But we can think a better way to let the user know when the installation is not completed.

adrivinca commented:
I have now added some documentation in the 'doc' folder.
and also slightly expanded the documentation of the `retixmp` package itself.

finally, the transport tutorials are now updated using new functionalities (loading entire dataframes).

gidden commented:
when this passes, I'll merge @adrivinca so it can get into our next release =)
